frontend/model/contracts/group.js

```diff
@@ -840,12 +840,16 @@ sbp('chelonia/defineContract', {
           const payment = state.payments[data.paymentHash]
 
           if (loggedIn.identityContractID === payment.data.toMemberID) {
-            sbp('gi.notifications/emit', 'PAYMENT_RECEIVED', {
-              createdDate: meta.createdDate,
-              groupID: contractID,
-              creatorID: innerSigningContractID,
-              paymentHash: data.paymentHash,
-              amount: getters.withGroupCurrency(payment.data.amount)
+            sbp('chelonia/queueInvocation', contractID, () => {
+              sbp('chelonia/queueInvocation', innerSigningContractID, () => {
+                sbp('gi.notifications/emit', 'PAYMENT_RECEIVED', {
+                  createdDate: meta.createdDate,
+                  groupID: contractID,
+                  creatorID: innerSigningContractID,
+                  paymentHash: data.paymentHash,
+                  amount: getters.withGroupCurrency(payment.data.amount)
+                })
+              })
```

Comment:
1. That's a good idea to make a new contract function like `emitAfterSyncing`. Then, I think I should add a comment at the top of the function.
2. About this approach (that uses several `chelonia/queueInvoation`), I had a discussion with @corrideat in [here](https://okturtles.slack.com/archives/C02QF71TK1Q/p1714988249312109?thread_ts=1714724867.814369&cid=C02QF71TK1Q). If we make a new chelonia function like `chelonia/contract/waitInSequence`, I think I can use that function. Should I make the new chelonia function in this PR?

Reply:
> About this approach (that uses several chelonia/queueInvoation), I had a discussion with @corrideat in [here](https://okturtles.slack.com/archives/C02QF71TK1Q/p1714988249312109?thread_ts=1714724867.814369&cid=C02QF71TK1Q). If we make a new chelonia function like chelonia/contract/waitInSequence, I think I can use that function. Should I make the new chelonia function in this PR?

Well, I saw that discussion, but I don't think it's necessary to use `chelonia/queueInvoation` to emit the event. `chelonia/queueInvoation` is used instead of `wait` when you want to make sure that an event is added to the event queue and that no other asynchronous events can happen until that event is done executing.

In this case it doesn't matter because we're just calling `'gi.notifications/emit'`, which has no impact on either of the event queues. So it's not necessary to create `chelonia/contract/waitInSequence`, you can just use `/wait` to wait until both contract queues are done, and then emit the event.

Reply:
> `chelonia/queueInvoation` is used instead of wait when you want to make sure that an event is added to the event queue and that no other asynchronous events can happen until that event is done executing. In this case it doesn't matter because we're just calling 'gi.notifications/emit', which has no impact on either of the event queues. 

We need it. That's because according to the updated workflow of joining group, we start syncing the identity contracts for the other group members after we fully finished the syncing group contracts. That's because the following codes are in the `sbp('chelonia/queueInvocation', contractID, () => {})` in the `/inviteAccept/sideEffect #1190`.
```
sbp('chelonia/contract/retain', profileIds).catch((e) => {
  console.error('Error while syncing other members\' contracts at inviteAccept', e)
})
```
But based on the previous codebase, we emit the notifications while we are syncing the group contracts at the time of we are ignorance of the group member's identity contracts. If you don't understand what I mean, you could try the following steps in **master** branch.
- [Tab1] user1 creates a group
- [Tab2] user2 joins the group
- [Tab3] user1 logs in the group, and has the weird notification with identity contract ID, instead of username

I think the best solution to handle notification text is that we should not generate the **text** at the time of emitting notification. The notifications should be saved as an **object format**, and they should be rendered in the `NotificationList.vue` or any other necessary places. That's because if we generate the notification text at the time of its creation, they are not updated to the user's profile, exactly the displayName at the moment. So we could have the following notifications for the same user `alexjin`.
```
The group has a new member. Say hi to alexjin!
Alex Jin proposed to add a member to the group. Vote now!
Alex created a proposal. Vote now!
SilverIT  proposed to change the group voting system. Vote now!
```
I am planning to create an issue for this issue this week.

Reply:
@Silver-IT ok, however, if that is what happens when using `/wait`, it will also happen when using `/queueInvocation`, because the timing will be the same. The only difference between using `/wait` with a `.then` (or an `await`), and `/queueInvocation`, is that `/queueInvocation` will have the function run on the queue instead of outside of it. Let me try to visualize this with a diagram:

```
Using /wait with groupID and identityContractID:

[--- groupID finishes syncing ---][---- identityContractID finishes syncing ---][-- some other event --]
                                                                                     [-- emit notification --]


Using /queueInvocation inside of identityContractID inside of groupID

[--- groupID finishes syncing ---][---- identityContractID finishes syncing ---][-- emit notification --][-- some other event --]
```

As you can see, `[-- emit notification --]` will happen after both contracts have finished syncing in both cases, and the only difference is that when using `/wait`, the notification might be emitted in the middle of some other event being processed on one of the queues.

> I am planning to create an issue for this issue this week.

Could you create and use the `'gi.contracts/group/emitAfterSyncing'` method in the meantime? It's fine to address any other issues in a future PR.

Reply:
> if that is what happens when using /wait, it will also happen when using /queueInvocation, because the timing will be the same.

@taoeffect, I don't think so. As you can see, `chelonia/contract/wait` uses `Promise.all` and all the `chelonia/queueInvocation` would be added **in parallel**, in other words **at the same time**.
```
'chelonia/contract/wait': function (contractIDs?: string | string[]): Promise<*> {
  const listOfIds = contractIDs
    ? (typeof contractIDs === 'string' ? [contractIDs] : contractIDs)
    : Object.keys(sbp(this.config.stateSelector).contracts)
  return Promise.all(listOfIds.flatMap(cID => {
    return sbp('chelonia/queueInvocation', cID, ['chelonia/private/noop'])
  }))
}
```
But, what I did is that `sbp('chelonia/queueInvocation', innerSigningContractID, () => {})` would run **after** the group contractID is **fully synced** because it is inside the queued invocation.

Reply:
OK, I see what you mean. Then, feel free then to either implement `chelonia/contract/waitInSequence`, or implement it directly inside of `'gi.contracts/group/emitAfterSyncing'`, e.g. something like:

```js
for (const id of contractIDs) {
  await sbp('chelonia/queueInvocation', id, ['chelonia/private/noop'])
}
// ... emit notification
```